### ISSUE TITLE
CI rerun fails with "Artifact * already exists for build #."

### ISSUE THREAD

**Issue by tannergooding** [dotnet/runtime#78427]
As an example, see https://github.com/dotnet/runtime/pull/76642/checks?check_run_id=9487437169

Where one of the legs was rerun but then failed with
> Build log #L33
Artifact mono_llvmfullaot_TestRunLogs_Linux_x64_release_innerloop already exists for build 82673.

**Comment by ViktorHofer** [comment]
Yes, that's a known issue on our side. We never had the bandwidth to fix it. Presumably for retries, we could fix this by appending the retry count to the asset id. Marking this issue with the "help wanted" label.
